### PR TITLE
JournalTCK, put delete tests behind a capability flag

### PR DESCRIPTION
**Pull request by chbatey**
Spanner emulator is currently broken for deletes so this will allow running all the other tests in travis against the emulator

### CODE REVIEW

File: akka-persistence-tck/src/main/scala/akka/persistence/CapabilityFlags.scala [hunk 1]
```diff
@@ -52,6 +52,11 @@ trait JournalCapabilityFlags extends CapabilityFlags {
    */
   protected def supportsSerialization: CapabilityFlag
 
+  /**
+   * When `true` enables tests which check if the journal supports deletes
+   */
+  protected def supportsDeletes: CapabilityFlag
```
Comment (johanandren) [inline akka-persistence-tck/src/main/scala/akka/persistence/CapabilityFlags.scala:58]: Default it to true to not break source compat for existing plugins?

Comment (chbatey) [reply akka-persistence-tck/src/main/scala/akka/persistence/CapabilityFlags.scala:58]: The override to true is in the JournalSpec

File: akka-persistence-tck/src/main/scala/akka/persistence/journal/JournalSpec.scala [hunk 2]
```diff
@@ -52,6 +52,7 @@ abstract class JournalSpec(config: Config)
   private var receiverProbe: TestProbe = _
 
   override protected def supportsSerialization: CapabilityFlag = true
+  override protected def supportsDeletes: CapabilityFlag = true
```
Comment (patriknw) [inline akka-persistence-tck/src/main/scala/akka/persistence/journal/JournalSpec.scala:55]: add same in SnapshotStoreSpec while you are at it

**Comment by chbatey** [comment]
Closing as i found a workaround for the emulator

**Pull request closed without merging**